misc/multistream-select/src/dialer_select.rs

```diff
@@ -55,44 +49,13 @@ pub fn dialer_select_proto<R, I>(
     protocols: I,
     version: Version,
 ) -> DialerSelectFuture<R, I::IntoIter>
-where
-    R: AsyncRead + AsyncWrite,
-    I: IntoIterator,
-    I::Item: AsRef<[u8]>,
-{
-    let iter = protocols.into_iter();
-    // We choose between the "serial" and "parallel" strategies based on the number of protocols.
-    if iter.size_hint().1.map(|n| n <= 3).unwrap_or(false) {
-        Either::Left(dialer_select_proto_serial(inner, iter, version))
-    } else {
-        Either::Right(dialer_select_proto_parallel(inner, iter, version))
-    }
-}
-
-/// Future, returned by `dialer_select_proto`, which selects a protocol and dialer
-/// either trying protocols in-order, or by requesting all protocols supported
-/// by the remote upfront, from which the first protocol found in the dialer's
-/// list of protocols is selected.
-pub type DialerSelectFuture<R, I> = Either<DialerSelectSeq<R, I>, DialerSelectPar<R, I>>;
-
-/// Returns a `Future` that negotiates a protocol on the given I/O stream.
-///
-/// Just like [`dialer_select_proto`] but always using an iterative message flow,
-/// trying the given list of supported protocols one-by-one.
-///
-/// This strategy is preferable if the dialer only supports a few protocols.
-pub(crate) fn dialer_select_proto_serial<R, I>(
-    inner: R,
-    protocols: I,
-    version: Version,
-) -> DialerSelectSeq<R, I::IntoIter>
 where
     R: AsyncRead + AsyncWrite,
     I: IntoIterator,
     I::Item: AsRef<[u8]>,
 {
     let protocols = protocols.into_iter().peekable();
-    DialerSelectSeq {
+    DialerSelectFuture {
         version,
         protocols,
         state: SeqState::SendHeader {
```

Comment:
We could also rename `SeqState` to `State` now.

Reply:
fixed